### PR TITLE
Add component for generate event attributes and event batching

### PR DESCRIPTION
**Pull request by yux0**
Add component for generating event attributes and event batching for event generator test framework. This is part of 3+DC

### CODE REVIEW

File: common/xdc/history_event_uitl.go [hunk 1]
```diff
@@ -0,0 +1,669 @@
...
+	buffer = buffer[:index]
+	truncatedBuffer = buffer
+	return
+}
+
+func shouldBatchWithDecisionTaskCompleted(event *shared.HistoryEvent) bool {
+	switch *event.EventType {
+	case shared.EventTypeWorkflowExecutionCompleted,
+		shared.EventTypeWorkflowExecutionContinuedAsNew,
+		shared.EventTypeWorkflowExecutionFailed,
```
Comment (wxing1292) [inline common/xdc/history_event_uitl.go]: missing EventTypeWorkflowExecutionCanceled

File: common/xdc/history_event_uitl.go [hunk 2]
```diff
@@ -0,0 +1,669 @@
...
+		shared.EventTypeActivityTaskScheduled,
+		shared.EventTypeActivityTaskCancelRequested,
+		shared.EventTypeTimerStarted,
+		shared.EventTypeTimerCanceled,
+		shared.EventTypeCancelTimerFailed,
+		shared.EventTypeRequestCancelExternalWorkflowExecutionInitiated,
+		shared.EventTypeMarkerRecorded,
+		shared.EventTypeStartChildWorkflowExecutionInitiated,
+		shared.EventTypeSignalExternalWorkflowExecutionInitiated:
```
Comment (wxing1292) [inline common/xdc/history_event_uitl.go]: missing EventTypeUpsertWorkflowSearchAttributes

File: common/xdc/history_event_uitl.go [hunk 3]
```diff
@@ -0,0 +1,669 @@
...
+		}
+		break
+	}
+	buffer = buffer[:index]
+	truncatedBuffer = buffer
+	return
+}
+
+func shouldBatchWithDecisionTaskCompleted(event *shared.HistoryEvent) bool {
+	switch *event.EventType {
```
Comment (wxing1292) [inline common/xdc/history_event_uitl.go]: GetEventType()

File: common/xdc/event_generator.go [hunk 4]
```diff
@@ -45,6 +45,7 @@ type (
 		GetNextVertex() []Vertex
 		ListGeneratedVertex() []Vertex
 		Reset()
+		RemoveVertexFromNextVertexCandidates(Vertex)
```
Comment (wxing1292) [inline common/xdc/event_generator.go]: plz add comments describing the terminologies, especially what does these means in the context of Cadence

File: common/xdc/history_event_uitl.go (hunk 1 above)
Comment (yux0) [reply common/xdc/history_event_uitl.go]: The batch logic moved inside the generator. https://github.com/uber/cadence/pull/2280

File: common/xdc/history_event_uitl.go (hunk 2 above)
Comment (yux0) [reply common/xdc/history_event_uitl.go]: The batch logic moved inside the generator. #2280

File: common/xdc/history_event_uitl.go (hunk 3 above)
Comment (yux0) [reply common/xdc/history_event_uitl.go]: The batch logic moved inside the generator. #2280

File: common/xdc/event_generator.go (hunk 4 above)
Comment (yux0) [reply common/xdc/event_generator.go]: Updated with more detailed doc.  #2280

**Comment by yux0** [comment]
Moved to #2280 due to commit disorder

**Pull request closed without merging**